tea.py

```diff
@@ -0,0 +1,443 @@
+#
+# TEA class
+# 
+
+import pybamm
+import matplotlib.pyplot as plt
+#import numpy as np
+import matplotlib.patches as patches
+import pandas as pd
+
+
+class TEA:
+    """
+    A Techno-Economic Analysis class for estimation of cell metrics:
+
+    Parameters
+    ----------
+    parameter_values : dict
+        A dictionary of parameters and their corresponding numerical values.
+        Default is NoneParameters
+    """
+
+    def __init__(self, parameter_values):
+        self.parameter_values = parameter_values
+
+    def print_stack_breakdown(self, print_dict=False):
+        if not hasattr(self, "stack_bd"):
```

Comment:
you could make `stack_breakdown` a property then when you call it will used the stored breakdown or call `calculate_stack_breakdown` first:
```python3
    @property
    def stack_breakdown(self):
        """A breakdown of ..."""
        try:
            return self._stack_breakdown
        except AttributeError:
            self._stack_breakdown = self.calculate_stack_breakdown()
            return self._t
```

Same for other stored calculations

Reply:
Then you just call `self.stack_breakdown` where you need it, and it will calculate it for you if it hasn't already.